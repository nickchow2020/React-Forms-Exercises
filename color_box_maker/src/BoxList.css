.boxesWrapper{
    display: flex;
    width: 80%;
    margin: auto;
    flex-direction: column;
}

.boxesContent{
    display: flex;
    flex-wrap: wrap;
}
