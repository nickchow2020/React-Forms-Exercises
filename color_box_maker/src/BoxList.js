import React,{useState} from "react"
import Box from "./Box"
import NewBoxForm from "./NewBoxForm"
import "./BoxList.css"
const BoxList = () =>{

    const [boxes,setBoxes] = useState([])

    const updateBoxes = (box)=>{
        setBoxes(boxes => ([...boxes,box]))
    }


    const deleteBox = (id) =>{
        setBoxes(boxes => (
            boxes.filter(box => box.id !== id)
        ))
    }


    return (
        <div className="boxesWrapper">
            <NewBoxForm updateBoxes={updateBoxes} />
            <div className="boxesContent">
                {
                    boxes.map( 
                        ({width, height,backgroundColor,id}) => <Box 
                        width={width} 
                        height={height} 
                        backgroundColor={backgroundColor} 
                        key={id}
                        id={id}
                        deleteBox={deleteBox}
                        />)
                }
            </div>
        </div>
    )
}   

export default BoxList