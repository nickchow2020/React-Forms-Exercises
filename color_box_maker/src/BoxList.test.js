import { getByTestId, render} from '@testing-library/react';
import BoxList from './BoxList';


test("should render BoxList",()=>{
    render(<BoxList />)
})


test("should match snapshot",()=>{
    const {asFragment} = render(<BoxList />)
    expect(asFragment()).toMatchSnapshot()
})


test("should display form",()=>{
    const {getByText} = render(<BoxList />)
    expect(getByText('Add New Box')).toBeInTheDocument()
})