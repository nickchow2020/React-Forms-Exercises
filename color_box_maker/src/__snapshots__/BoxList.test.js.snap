// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="boxesWrapper"
  >
    <form>
      <h1>
        Add New Box
      </h1>
      <label
        for="background"
      >
        Background_Color:
        <input
          data-testid="color"
          id="background"
          name="backgroundColor"
          placeholder="background color"
          type="text"
          value=""
        />
      </label>
      <label
        for="width"
      >
        Box_width:
        <input
          data-testid="width"
          id="width"
          name="width"
          placeholder="box width"
          type="text"
          value=""
        />
      </label>
      <label
        for="height"
      >
        Box_height:
        <input
          data-testid="height"
          id="height"
          name="height"
          placeholder="box height"
          type="text"
          value=""
        />
      </label>
      <button
        data-testid="submit"
      >
        Add Now Box
      </button>
    </form>
    <div
      class="boxesContent"
    />
  </div>
</DocumentFragment>
`;
